File: backend/TenderGo.Models/ENUMs/ApplicationStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TenderGo.Models.ENUMs
{
    public enum ApplicationStatus
    {
       Pending = 1,
        Accepted = 2,
        Rejected = 3,
        Withdrawn = 4,   
        Cancelled = 5
    }
}

public enum TenderStatus
{
    Open = 1,
    Closed = 2,
    Awarded = 3,
    Cancelled = 4
}

File: UI/tendergo_mobile/lib/shared/models/enums/application_status.dart
enum ApplicationStatus {
  pending(1),
  accepted(2),
  rejected(3),
  withdrawn(4),
  cancelled(5),;

  final int value;
  const ApplicationStatus(this.value);

  static ApplicationStatus fromValue(dynamic val) {
    if (val is int) {
      return ApplicationStatus.values.firstWhere((e) => e.value == val, orElse: () => pending);
    }
    if (val is String) {
      final str = val.toLowerCase().trim();
      return ApplicationStatus.values.firstWhere((e) => e.name.toLowerCase() == str, orElse: () => pending);
    }
    return pending;
  }
}